wave_optimization/include/wave/optimization/factor_graph/ValueView.hpp

```diff
@@ -0,0 +1,82 @@
+/**
+ * @file
+ * @ingroup optimization
+ */
+
+#ifndef WAVE_OPTIMIZATION_FACTOR_GRAPH_VALUE_HPP
+#define WAVE_OPTIMIZATION_FACTOR_GRAPH_VALUE_HPP
+
+#include <iostream>
+
+#include "wave/utils/math.hpp"
+#include "wave/optimization/factor_graph/FactorVariableBase.hpp"
+
+namespace wave {
+/** @addtogroup optimization
+ *  @{ */
+
+
+/*
+ * A structure mapping useful types to an underlying value buffer.
+ *
+ * While variables and values are fundamental to the factor graph approach, a
+ * _view_ is a syntactic convenience. It is a data structure which assigns
+ * labels to an underlying value, to simplify interacting with it. For example,
+ * a Pose3D view may define elements of the underlying 6D value as `position`
+ * and `orientation`. In the Factor's evaluation function, these objects can
+ * then be used as independent Eigen vectors.
+ *
+ * This class is meant to be derived by specialized variable types which assign
+ * meaning to the underlying data vector.
+ *
+ * @todo add more syntactic sugar
+ */
+template <int S>
+class ValueView {
+ public:
+    constexpr static int Size = S;
```

Comment:
Why not just set this as the template variable name. i.e. `template <int Size>`.

Sorry if I'm missing some style guide thing here. 😅